jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -7526,9 +7549,12 @@ private static final boolean hasMetaChars(String s) {
 	 * @since 1.4
 	 */
 	public String[] split(String regex, int max) {
-		// it is faster to handle simple splits inline (i.e. no fancy regex matching)
+		// it is faster to handle simple splits inline (i.e. no fancy regex matching),
+		// including single escaped literal character (e.g. \. \{),
 		// so we test for a suitable string and handle this here if we can
-		if (regex != null && regex.lengthInternal() > 0 && !hasMetaChars(regex)) {
+		boolean singleEscapeLiteral = false;
+		if (regex != null && regex.lengthInternal() > 0 
+				&& (!hasMetaChars(regex) || (singleEscapeLiteral = isSingleEscapeLiteral(regex)))) {
```

Comment:
This is doing the assignment inside the if statement.  Please move the assignment to its own line.

Reply:
Also the formatting here is wonky - see the comment on the version of this code above